### ISSUE TITLE
Allow direct download for Oruxmap

### ISSUE THREAD

**Issue by zstadler** [IsraelHikingMap/Site#133]
Ready-to-download Israel Hiking map for Oruxmaps is available at Map\Site\Oruxmaps but it is not yet accessible.

Todo:
* Make the directory accessible
* Make .db and .xml files accessible

It will also be a more reliable way to download the map for publishing to the current Google Drive location

**Comment by HarelM** [comment]
Sure thing, these are mainly web.config settings that needs to be added.

**Comment by zstadler** [comment]
The ready to use maps are now available at http://israelhiking.osm.org.il/Oruxmaps/

Making the directory and files accessible was done in two steps:
1. Allowing "directory listing" for the parent "Oruxmaps" directory in IIS
2. Providing the IsraelHiking.org.il application pool account with read access to the directory tree, following the [detailed instructions at here](http://kc.litiumstudio.se/documentation/litium-studio/tips-tricks/give-directory-access-permission-to-iis-application-pool)

**Comment by HarelM** [comment]
@zstadler I have added a small piece of code that allows directory listing that looks very nice.
you can have a look at:
http://israelhiking.osm.org.il/files/
It takes its physical folder from the web.config file under entry directoryListing.
it's currently pointing to C:\Temp.
I can change the code to have whichever extension to the address and to which ever physical path.
Please update me what you would like to list here.

**Comment by HarelM** [comment]
@zstadler any updates on this? it is waiting on you in order to be closed...

**Comment by zstadler** [comment]
This is not at our control. It may be supported in a future Oruxmaps
release

**Comment by HarelM** [comment]
I'm referring to the directory listing I've added, see my post above...

**Comment by zstadler** [comment]
Please apply it to http://israelhiking.osm.org.il/Oruxmaps/ so we can check its compatibility with ```wget --recusive```.
If it works well, we should also apply it ti Tiles, MTBTiles, and OverlayTiles

**Comment by HarelM** [comment]
@zstadler Done, Lets close this one and open another issue for oruxmaps app?
(I'm extremely proud of how it turned out :-)).

**Comment by zstadler** [comment]
Great!!!

When you close this issue please edit its title accordingly.

By the way, is there a way to add sort options like this site
<http://download.geofabrik.de/asia/>?

**Comment by zstadler** [comment]
The new mechanism now shows hidden files, like web.config files, that were not shown by the default directory listing mechanism.

Can you add this filter?

**Comment by HarelM** [comment]
I'll see what I can do but I won't get my hopes up since I'm using basic C# operations like get all files from  directory without consulting the IIS if they are available...

**Comment by zstadler** [comment]
As far as I understand, this is not about asking the permission IIS has to the files, but rather asking the file system if a file is hidden.

Currently the listing looks nicer, but it shows file that will produce a 404 error. If this issue cannot be fixed, I prefer the original, less attractive but fully functional listing.

**Comment by HarelM** [comment]
web.config is not a hidden file but original listing does hide it.
I'll see if I can make it better...

**Comment by HarelM** [comment]
@zstadler Please let me know if the fix I made suites your needs.
Not entirely related, where do you think the link to this folder/instruction should be added?
Do we still want to obscure it?

**Comment by zstadler** [comment]
I didn't understand the comment that "web.config is not a hidden file".
Regardless, the implementation indeed hides hidden files, as requested.

Note that the title of all listing is "Index of /" whether it lists 
http://israelhiking.osm.org.il/Oruxmaps/, http://israelhiking.osm.org.il/Tiles/, http://israelhiking.osm.org.il/mtbTiles/, or http://israelhiking.osm.org.il/OverlayTiles/

**Comment by HarelM** [comment]
Done, Documentation update is needed, Also adding a link in the site it self would probably help.

**Comment by zstadler** [comment]
The titles of some pages create incorrect, yet working, links. It confuses the recursive tiles update process using `wget` and causes it to create many top-level empty directories for all lower level directories.

For example, the word "9785/" at the top of the http://israelhiking.osm.org.il/Tiles/14/9785/ page is a link to http://israelhiking.osm.org.il/Tiles/9795/ - the "14" was omitted. This link shows an empty directory and as a result, `wget` would create the empty directory `Tiles\9795`:

`2016-04-13 08:57:34 URL:http://israelhiking.osm.org.il/Tiles/9785/ [2765/2765] -> "Tiles/9785/index.html" [1]`

I see here two issues:
1. The link is incorrect
2. There is no error when trying to access a non-existing directory, such as http://israelhiking.osm.org.il/Tiles/FOO/, as opposed to a 404 error for http://israelhiking.osm.org.il/FOO/

**Comment by HarelM** [comment]
Nice catch!
